src/github.com/couchbase/sync_gateway/db/design_doc.go

```diff
@@ -110,34 +110,47 @@ func (db *Database) QueryDesignDoc(ddocName string, viewName string, options map
 			}
 		}
 	} else {
-		result = filterViewResult(result, db.user)
+		result = filterViewResult(result, db.user, options["reduce"] == true)
 	}
 	return &result, nil
 }
 
 // Cleans up the Value property, and removes rows that aren't visible to the current user
-func filterViewResult(input sgbucket.ViewResult, user auth.User) (result sgbucket.ViewResult) {
+func filterViewResult(input sgbucket.ViewResult, user auth.User, reduce bool) (result sgbucket.ViewResult) {
 	checkChannels := false
 	var visibleChannels ch.TimedSet
 	if user != nil {
 		visibleChannels = user.InheritedChannels()
 		checkChannels = !visibleChannels.Contains("*")
+		if (reduce) {
```

Comment:
Does this PR intentionally not address the case where a non-admin user queries a reduce view, but doesn't set reduce=true in the request?  That will just continue to fail in the same way it fails today?